src/drawscene.cpp

```diff
@@ -459,21 +459,58 @@ void draw_pageflip_3d_mode(Camera& camera, bool show_hud,
 #endif
 }
 
-void draw_plain(Camera &camera, bool show_hud, Hud &hud,
-		video::IVideoDriver *driver, bool draw_wield_tool,
-		Client &client, gui::IGUIEnvironment *guienv)
+// returns (size / coef), rounded upwards
+inline int scaledown(int coef, int size)
 {
+	return (size + coef - 1) / coef;
+}
+
+void draw_plain(Camera &camera, bool show_hud,
+		Hud &hud, video::IVideoDriver *driver,
+		scene::ISceneManager *smgr, const v2u32 &screensize,
+		bool draw_wield_tool, Client &client, gui::IGUIEnvironment *guienv,
+		video::SColor skycolor)
+{
+	// Undersampling-specific stuff
+	static video::ITexture *image = NULL;
+	static v2u32 last_pixelated_size = v2u32(0, 0);
+	int undersampling_x = g_settings->getU16("undersampling_x");
+	int undersampling_y = g_settings->getU16("undersampling_y");
+	bool undersampling_enabled = (undersampling_x > 0) || (undersampling_y > 0);
```

Comment:
This should be `&&` to not get a division by zero in `scaledown` if only one value is set.